src/Solvers/multigrid_solver.jl

```diff
@@ -307,29 +304,45 @@ end
     s = solver.amgx_solver
     AMGX.upload!(s.device_b, solver.b_array)
     AMGX.upload!(s.device_x, solver.x_array)
-    AMGX.setup!(s.solver, s.device_matrix)
     AMGX.solve!(s.device_x, s.solver, s.device_b)
     AMGX.copy!(solver.x_array, s.device_x)
 
     interior(x) .= reshape(solver.x_array, Nx, Ny, Nz)
 end
 
-@ifhasamgx function finalize_solver!(solver::MultigridGPUSolver)
-    @info "Finalizing the Multigrid solver on GPU"
-    
-    s = solver.amgx_solver
+initialize_AMGX(::CPU) = nothing
```

Comment:
Perhaps add docstrings for `initialize_AMGX` and `finalize_AMGX`? Also, are these exported?

Reply:
Sounds good. I don't really understand exporting - why would I export something if it can already be called from the `benchmark` directory?

Reply:
Export means that the method is available in the main Oceananigans namespace. E.g., at the moment

```Julia
julia> using Oceananigans

julia> initialize_AMGX()
ERROR: UndefVarError: initialize_AMGX not defined
Stacktrace:
 [1] top-level scope
   @ REPL[2]:1
 [2] top-level scope
   @ ~/.julia/packages/CUDA/DfvRa/src/initialization.jl:52
```

But perhaps this is specialized enough so we can leave it _hidden_, and only available when 

```Julia
julia> using Oceananigans

julia> using Oceananigans.Solvers: initialize_AMGX

julia> initialize_AMGX(CPU())
```